src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -160,3 +156,26 @@ export const updateEntityId: ActionCreator<ThunkAction<
     dispatch(updateActiveEntityForGenome(updatedActiveEntityIds));
   };
 };
+
+export const deleteGenome = createAction('entity-viewer/delete-genome')<
+  string
+>();
+
+export const deleteSpeciesInEntityViewer = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    dispatch(deleteGenome(genomeIdToRemove));
+
+    const state = getState();
+
+    const updatedActiveEntityIds = pickBy(
+      getEntityViewerActiveEntityIds(state),
+      (value, key) => key !== genomeIdToRemove
+    );
+
+    dispatch(updateActiveEntityForGenome(updatedActiveEntityIds));
```

Comment:
I know we've had similar conversations before, but I still think that the logic that's happening here on lines 170-177 is better suited for reducers. The reducers that are interested in `entityViewerGeneralActions.deleteGenome` action will react to it by removing what's related to the deleted genome id from the state slices that they are responsible for.

Reply:
That's done. These lines were not required as the reducer already does a similar task.